discovery_engine_core/core/src/storage/sqlite/dart_migrations.rs

```diff
@@ -29,13 +31,25 @@ pub(super) async fn store_migration_data(
     // it's okay to not have an transaction across the various migrations:
     // 1. by taking `&mut SqliteStorage` we know we have exclusive access
     // 2. databases of failed migrations should be discarded at some point
-    // 3. even if the database is not discarded the db is still in a valid state,
+    // 3. even if the database is not discarded the db is still in a sound state,
     //    just with some history/config/preference or similar missing
 
     if let Some(engine_state) = &data.engine_state {
         storage.state().store(engine_state).await?;
     }
 
+    storage
+        .source_preference()
+        .set_trusted(&data.trusted_sources.iter().map_into().collect())
+        .await
+        .unwrap();
```

Comment:
is there a chance for `panic` here? if not maybe a comment why is it safe could be useful here